### ISSUE TITLE
how to update Searx instance installed with Docker?

### ISSUE THREAD

**Issue by nixoye** [searx/searx#680]
I've been using my Searx instance for a while and would like to update to 0.10 but I'm not sure whats the best way to go about updating it without breaking something

**Comment by Wonderfall** [comment]
Rebuild the image and recreate the container, it's as simple as that.

**Comment by nixoye** [comment]
@Wonderfall i ran 

docker build -t nix/searx .

then did 
docker stop searx && docker rm searx

and then try 

docker run -d --name searx -p 80:8888 nix/searx

but it doesnt run. If i check docker ps -a it says CREATED 14 seconds ago Exited 11 seconds ago

**Comment by Wonderfall** [comment]
What is the output of `docker logs searx` ?

**Comment by nixoye** [comment]
> sed: searx.setting.yml: No such file or directory
> Traceback (most recent call last):
>   File "searx/webapp.py", line 826, in <module>
>     run()
>   File "searx/webapp.py", line 781, in run
>     host=settings['server']['bind_address']
>   File "/usr/lib/python2.7/site-packages/flask/app.py", line 772, in run
>     run_simple(host, port, self, *_options)
>   File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 694, in run_simple
>     inner()
>   File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 656, in inner
>     fd=fd)
>   File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 550, in make_server
>     passthrough_errors, ssl_context, fd=fd)
>   File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 464, in **init**
>     HTTPServer.**init**(self, (host, int(port)), handler)
>   File "/usr/lib/python2.7/SocketServer.py", line 417, in **init**
>     self.server_bind()
>   File "/usr/lib/python2.7/BaseHTTPServer.py", line 108, in server_bind
>     SocketServer.TCPServer.server_bind(self)
>   File "/usr/lib/python2.7/SocketServer.py", line 431, in server_bind
>     self.socket.bind(self.server_address)
>   File "/usr/lib/python2.7/socket.py", line 228, in meth
>     return getattr(self._sock,name)(_args)
> socket.error: [Errno 13] Permission denied
> sed: searx.setting.yml: No such file or directory
> Traceback (most recent call last):
>   File "searx/webapp.py", line 826, in <module>
>     run()
>   File "searx/webapp.py", line 781, in run
>     host=settings['server']['bind_address']
>   File "/usr/lib/python2.7/site-packages/flask/app.py", line 772, in run
>     run_simple(host, port, self, *_options)
>   File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 694, in run_simple
>     inner()
>   File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 656, in inner
>     fd=fd)
>   File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 550, in make_server
>     passthrough_errors, ssl_context, fd=fd)
>   File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 464, in **init**
>     HTTPServer.**init**(self, (host, int(port)), handler)
>   File "/usr/lib/python2.7/SocketServer.py", line 417, in **init**
>     self.server_bind()
>   File "/usr/lib/python2.7/BaseHTTPServer.py", line 108, in server_bind
>     SocketServer.TCPServer.server_bind(self)
>   File "/usr/lib/python2.7/SocketServer.py", line 431, in server_bind
>     self.socket.bind(self.server_address)
>   File "/usr/lib/python2.7/socket.py", line 228, in meth
>     return getattr(self._sock,name)(_args)
> socket.error: [Errno 13] Permission denied
> sed: searx.setting.yml: No such file or directory
> Traceback (most recent call last):
>   File "searx/webapp.py", line 826, in <module>
>     run()
>   File "searx/webapp.py", line 781, in run
>     host=settings['server']['bind_address']
>   File "/usr/lib/python2.7/site-packages/flask/app.py", line 772, in run
>     run_simple(host, port, self, *_options)
>   File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 694, in run_simple
>     inner()
>   File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 656, in inner
>     fd=fd)
>   File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 550, in make_server
>     passthrough_errors, ssl_context, fd=fd)
>   File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 464, in **init**
>     HTTPServer.**init**(self, (host, int(port)), handler)
>   File "/usr/lib/python2.7/SocketServer.py", line 417, in **init**
>     self.server_bind()
>   File "/usr/lib/python2.7/BaseHTTPServer.py", line 108, in server_bind
>     SocketServer.TCPServer.server_bind(self)
>   File "/usr/lib/python2.7/SocketServer.py", line 431, in server_bind
>     self.socket.bind(self.server_address)
>   File "/usr/lib/python2.7/socket.py", line 228, in meth
>     return getattr(self._sock,name)(_args)
> socket.error: [Errno 13] Permission denied

**Comment by fmbento** [comment]
Consider removing the image before its rebuild ("docker rmi" of it -- in
case there's some entry that dockers considers to be already covered by an
existing layer of that image).

2016-09-07 0:15 GMT+01:00 NIXOYE notifications@github.com:

**Comment by NIXOYE** [comment]
@fmbento so i do 'docker rmi (docker number)' ?

@Wonderfall any thoughts on the logs?

**Comment by nixoye** [comment]
so apparently my problem was that i was trying to run it while root user. I got my 0.9 version going again but i still cant update to 0.10

@Wonderfall could you ELI5 how to rebuild the image and the container?

**Comment by fmbento** [comment]
Sorry, missed this reply of yours.

   docker rmi <image name> or <id>

Get the image id via

  docker images

Image name would be, in your case, nix/searx

Clean up all the <none>:<none>

  docker rmi $(docker images -q -f dangling=true)

Hope this helps, at least to free some space. Please do let us know how it
goes.

Thanks,
Filipe

2016-09-07 20:24 GMT+01:00 NIXOYE notifications@github.com:

**Comment by nixoye** [comment]
@fmbento I tried this and got to get rid of a bunch of docker images that were taking up space (thanks) but it still didn't work. I ended up deleting everything in the searx folder with rm -f searx and then re installed searx with the install instructions at https://github.com/asciimoo/searx/wiki/Installation

**Comment by Pofilo** [comment]
A new docker repo can help you: https://github.com/searx/searx-docker.
